src/components/Viewer/context/OperationCtx.tsx

```diff
@@ -1,19 +1,20 @@
-import { Operation, Options } from 'math/operations';
+import { Operation } from 'math/operations';
 import { createHookedContext } from 'components/common';
 
 const defaultState = {
   operation: undefined,
   options: undefined,
 };
 
-interface State {
-  operation?: Operation;
+type Options = Record<string, any>;
+interface State<Options extends {}> {
+  operation?: Operation<Options>;
   options?: Options;
 }
 
 type Actions = 'setOperation' | 'unsetOperation' | 'setOption';
 
-export default createHookedContext<State, Actions>(
+export default createHookedContext<State<Options>, Actions>(
```

Comment:
What's the best way to handle cases where we're pulling in all these types of `Options`? Maybe make `DefaultOptions = Record<string, any>` into the default value.